integration-tests/smoke/keeper_test.go

```diff
@@ -642,6 +645,65 @@ var _ = Describe("Keeper Suite @keeper", func() {
 			}, "1m", "1s").Should(Succeed())
 		}
 
+		if testToRun == HandleKeeperNodesGoingDown {
+			By("takes down half of the keeper nodes and watches upkeeps get performed slower")
+			// Record how much time it takes to have all the registered upkeeps perform 5 times each (the first time)
+			const numberOfTimesUpkeepsShouldPerform int64 = 5
+
+			var completedUpkeepsFirstTime = 0
+			var alreadyMarkedFirstTime = make([]bool, len(upkeepIDs))
+
+			firstStart := time.Now()
+			for completedUpkeepsFirstTime < len(upkeepIDs) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					counter, err := consumers[i].Counter(context.Background())
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to get counter for upkeepID"+strconv.Itoa(i))
+
+					if counter.Cmp(big.NewInt(numberOfTimesUpkeepsShouldPerform)) == 0 && !alreadyMarkedFirstTime[i] {
+						completedUpkeepsFirstTime++
+						alreadyMarkedFirstTime[i] = true
+					}
+				}
+			}
+
+			firstDuration := time.Since(firstStart)
+			log.Info().Msg("Executing each upkeep " + strconv.Itoa(int(numberOfTimesUpkeepsShouldPerform)) +
+				" amount of times for the first time took " + firstDuration.String())
+
+			// Now we need to take down half of the keeper nodes
+			nodesToTakeDown := chainlinkNodes[:len(chainlinkNodes)/2+1]
+			for _, nodeToTakeDown := range nodesToTakeDown {
+				err := nodeToTakeDown.DeleteJob("1")
+				Expect(err).ShouldNot(HaveOccurred(), "Could not delete the job from the node")
+			}
+			log.Info().Msg("Successfully managed to take down half of the nodes")
+
+			// Record how much time it takes to have all the registered upkeeps perform 5 times each (the second time)
+			var completedUpkeepsSecondTime = 0
+			var alreadyMarkedSecondTime = make([]bool, len(upkeepIDs))
+
+			secondStart := time.Now()
+			for completedUpkeepsSecondTime < len(upkeepIDs) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					counter, err := consumers[i].Counter(context.Background())
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to get counter for upkeepID"+strconv.Itoa(i))
+					if counter.Cmp(big.NewInt(numberOfTimesUpkeepsShouldPerform*2)) == 0 && !alreadyMarkedSecondTime[i] {
+						completedUpkeepsSecondTime++
+						alreadyMarkedSecondTime[i] = true
+					}
+				}
+			}
+
+			secondDuration := time.Since(secondStart)
+			log.Info().Msg("Executing each upkeep " + strconv.Itoa(int(numberOfTimesUpkeepsShouldPerform)) +
+				" amount of times for the second time took " + secondDuration.String())
+
+			// Make sure that the second time it took significantly longer to execute
+			// each upkeep the given amount of times
+			Expect(secondDuration.Seconds()-firstDuration.Seconds()).Should(BeNumerically(">", 20),
+				"Expected the second measurement to be considerably slower than the first")
+		}
```

Comment:
this seems like it could cause flakiness due to varying infra conditions. do we really need this expect? Maybe it's enough to check that upkeeps are being executed.

Reply:
That's a good idea! Without any more time measurements, right? Just focusing on whether the upkeeps are getting performed or not once the other half has been taken down.